app/controllers/docs_controller.rb

```diff
@@ -7,23 +7,29 @@ class DocsController < ApplicationController
   def index; end
 
   def section
-    @doc = Document.find_by(section: @section, slug: "APEX")
+    @doc = Document.find_by!(section: @section, slug: "APEX")
     render action: :show
+  rescue ActiveRecord::RecordNotFound
+    render_404
```

Comment:
Out of interest, do you need this? I'd have guessed Rails would gracefully handle the `ActiveRecord::RecordNotFound` to show a 404.

Reply:
You are correct. I've checked this by changing `config.consider_all_requests_local = true` to `config.consider_all_requests_local = false` in `config/environments/development.rb`.

The only "downside" is that the tests will change from

```ruby
get docs_section_path(:mentoring)
assert_rendered_404
```

to:

```ruby
assert_raises ActiveRecord::RecordNotFound do
  get docs_section_path(:mentoring)
end
```

I like the former a bit better.

Reply:
So, I think it's definitely not idiomatic to be catching ActiveRecord::RecordNotFound and reraising as 404s, so we shouldn't do it.

I agree about the tests though. I'm surprised the first doesn't work tbh. I'd have thought it would have done reflected the production output more. But I think we probably just have to live with the uglier tests.

Reply:
> I agree about the tests though. I'm surprised the first doesn't work tbh.

Well, that's because `environments/test.rb` has the same define:

```ruby
config.consider_all_requests_local = true
```

Do we want that?

Reply:
Yeah, I imagine we should leave it as it is. But I just figured Rails would be smarter, despite that :)